### PR TITLE
Add support for @oneOf directives in printSchema

### LINKED ISSUE #3968: Printing the @oneOf directive
It looks like currently the `printSchema` function doesn't have any handling for `@oneOf`.

I'm not sure if this is intentional or not, because I know printing directives has historically been a controversial topic, but my expectation would be that for directives that are part of the spec and have a corresponding option in the options when creating the type/field in code (eg `deprecationReason`/`@deprecated` ) would be printed when using  `printSchema`.

I'm happy to make a PR to add this, and it looks like it should be a pretty simple fix handled here: https://github.com/graphql/graphql-js/blob/main/src/utilities/printSchema.ts#L207C5-L207C5

### PR DESCRIPTION
**Pull request by hayes**
fixes #3968

### CODE REVIEW

File: src/utilities/__tests__/printSchema-test.ts [hunk 1]
```diff
@@ -663,7 +680,7 @@ describe('Type System Printer', () => {
       schema {
         query: Query
       }
-      
+
```
Comment (IvanGoncharov) [inline src/utilities/__tests__/printSchema-test.ts:683]: Thanks for spotting it!
I applied global fix, here: https://github.com/graphql/graphql-js/pull/3970

File: src/utilities/printSchema.ts [hunk 2]
```diff
@@ -204,7 +204,12 @@ function printInputObject(type: GraphQLInputObjectType): string {
   const fields = Object.values(type.getFields()).map(
     (f, i) => printDescription(f, '  ', !i) + '  ' + printInputValue(f),
   );
-  return printDescription(type) + `input ${type.name}` + printBlock(fields);
+  return (
+    printDescription(type) +
+    `input ${type.name}` +
+    printOneOf(type.isOneOf) +
```
Comment (IvanGoncharov) [inline src/utilities/printSchema.ts:210]: nit: It's overkill to create separate function, ternary operator would be enough here: `type.isOneOf ? ' @oneOf' : ''`.

Comment (hayes) [reply src/utilities/printSchema.ts]: makes sense, was just trying to match the existing pattern for the deprecated directive (which is slightly more complex).  Updated!

**Comment by hayes** [comment]
anything else you need from my end to get this merged?

**Review by benjie — APPROVED** [review]
Thanks for spotting this omission @hayes; this is critical!

**Comment by benjie** [comment]
@github-actions publish-pr-on-npm

**Comment by benjie** [comment]
@graphql/graphql-js-reviewers This seems like an easy merge now all the feedback has been addressed? :raised_hands:

**Review by yaacovCR — APPROVED** [review]

**Review by saihaj — APPROVED** [review]

**Pull request merged**